packages/core/src/modules/presentation-exchange/PresentationExchangeService.ts

```diff
@@ -416,13 +461,14 @@ export class PresentationExchangeService {
       }
 
       // Clients MUST ignore any presentation_submission element included inside a Verifiable Presentation.
-      const presentationToSign = { ...presentationJson, presentation_submission: undefined }
+      const presentationToSign = { ...presentationJson }
+      delete presentationToSign['presentation_submission']
 
       let signedPresentation: W3cVerifiablePresentation<ClaimFormat.JwtVp | ClaimFormat.LdpVp>
       if (vpFormat === 'jwt_vp') {
         signedPresentation = await w3cCredentialService.signPresentation(agentContext, {
           format: ClaimFormat.JwtVp,
-          alg: this.getSigningAlgorithmForJwtVc(presentationDefinition, verificationMethod),
+          alg: this.getSigningAlgorithmForJwtVc(presentationDefinition as PresentationDefinition, verificationMethod),
```

Comment:
Is presentationDefinition possibly undefined? Can we do a check in that case? Then we don't have to use type casting